internal/httputil/httputil.go

```diff
@@ -5,30 +5,60 @@ import (
 	"crypto/x509"
 	"net/http"
 	"os"
+	"path/filepath"
+	"strings"
 	"time"
 )
 
-func BuildHTTPClient(caCert string) (*http.Client, error) {
-	httpClient := &http.Client{Timeout: 10 * time.Second}
-
-	if caCert != "" {
-		// tlsFileWatcher, err := certwatcher.New(caCert, "")
+func LoadCerts(caDir string) (string, error) {
```

Comment:
Seems like it would be a nicer abstraction if this was:
```suggestion
func AddCertificatesFromDirectory(certPool *x509.CertPool, caDir string) error { ... }
```
In which case, we could add each file's content, collect errors reading/adding each file, and then tell callers which specific files failed for which reasons.

Reply:
This is not compatible with the current rukpak API, which expects a string of PEM. Maybe when rukpak is fully integrated into operator-controller? I agree that this would be better for a larger directory. There does not appear to be a way to extract certificates/PEM from a CertPool, so it can't be done with this signature, yet.

Reply:
Oh I was just looking at `LoadCerts` being called in `BuildHTTPClient`, which turns around and consumes the `certs` string by immediately adding to the `caCertPool`.

And that made me think we could pass the `caCertPool` into this function and call `AppendCertsFromPEM` for each file's contents. Then, when my suggested function signature returns, you end up with `caCertPool` having everything in `caDir` added to it.

Seems like I'm missing something though?

Reply:
It's also used when creating a BundleDeployment for a rukpak API.
https://github.com/operator-framework/operator-controller/blob/ed4e40eaca385c8bd4d1d7e8329c9f9c76013d95/internal/controllers/clusterextension_controller.go#L252

So, we need to change the rukpak API to accept a caPool first, then we can do as you suggest.